webapp/channels/src/components/channel_header_menu/channel_header_menu.tsx

```diff
@@ -0,0 +1,184 @@
+                onClick={() => {
+                    if (item.action) {
+                        item.action(channel.id);
+                    }
+                }}
+                labels={<span>{item.text}</span>}
+            />
+        );
+    });
+
+    return (
+        <Menu.Container
+            menuButtonTooltip={{
+                text: channelTitle as string,
+            }}
+            menuButton={{
+                id: 'channelHeaderDropdownButton',
+                class: classNames('channel-header__trigger style--none'),
+                children: (
+                    <>
+                        {archivedIcon}
+                        <strong
+                            id='channelHeaderTitle'
+                            className='heading'
+                            role='heading'
+                            aria-level={2}
+                        >
+                            {channelTitle as string}
+                        </strong>
+                        {sharedIcon}
+                        <ChevronDownIcon size={16}/>
+                    </>
+                ),
+                'aria-label': ariaLabel.toLowerCase(),
+            }}
+            menu={{
+                id: 'channelHeaderDropdownMenu',
+            }}
+            transformOrigin={{
+                horizontal: 'left',
+                vertical: 'top',
+            }}
+            anchorOrigin={{
+                vertical: 'bottom',
+                horizontal: 'left',
+            }}
+        >
+            {isDirect && (
+                <ChannelDirectMenu
+                    channel={channel}
+                    user={user}
+                    isMuted={isMuted}
+                    pluginItems={pluginItems}
+                    isFavorite={isFavorite}
+                    isMobile={isMobile || false}
+                />
+            )}
+            {isGroup && (
+                <ChannelGroupMenu
+                    channel={channel}
+                    user={user}
+                    isMuted={isMuted}
+                    pluginItems={pluginItems}
+                    isFavorite={isFavorite}
+                    isMobile={isMobile || false}
+                />
+            )}
+            {(!isDirect && !isGroup) && (
+                <ChannelPublicPrivateMenu
+                    channel={channel}
+                    user={user}
+                    isMuted={isMuted}
+                    pluginItems={pluginItems}
+                    isFavorite={isFavorite}
+                    isMobile={isMobile || false}
+                    isReadonly={isReadonly}
+                    isDefault={isDefault}
+                    isLicensedForLDAPGroups={isLicensedForLDAPGroups}
+                />
+            )}
+
+            {isMobile && pluginItems}
+            {isMobile && (
+                <MobileChannelHeaderPlugins
+                    channel={channel}
+                    isDropdown={true}
+                />
+            )}
```

Comment:
We can further separate these into a new component, ChannelHeaderMobileMenu, and include both of the above menu items within it. This way, we only need to check whether it’s a mobile view inside this component, and we can lazily import ChannelHeaderMobileMenu.